protocol/clproto_cpp/src/decoders.cpp

```diff
@@ -1,35 +1,74 @@
 #include "clproto/decoders.h"
 
-#include <state_representation/State.hpp>
-
-#include "state_representation/state.pb.h"
-#include "state_representation/space/cartesian/cartesian_state.pb.h"
-#include "state_representation/space/cartesian/cartesian_state.pb.h"
+#include "state_representation/state_message.pb.h"
 
 using namespace state_representation;
 
 namespace clproto {
 
 DecoderNotImplementedException::DecoderNotImplementedException(const std::string& msg) : DecodingException(msg) {}
 
+template<>
+std::vector<bool> decoder(const google::protobuf::RepeatedField<bool>& message) {
+  // explicit construction is needed for the bool vector due to stl optimisations
+  std::vector<bool> vec(message.begin(), message.end());
+  return vec;
+}
+
 template<>
 StateType decoder(const proto::StateType& message) {
   return static_cast<StateType>(message);
 }
 
 template<>
 Eigen::Vector3d decoder(const proto::Vector3d& message) {
-  return Eigen::Vector3d(message.x(), message.y(), message.z());
+  return {message.x(), message.y(), message.z()};
 }
 
 template<>
 Eigen::Quaterniond decoder(const proto::Quaterniond& message) {
-  return Eigen::Quaterniond(message.w(), message.vec().x(), message.vec().y(), message.vec().z());
+  return {message.w(), message.vec().x(), message.vec().y(), message.vec().z()};
```

Comment:
why did you change here? Also I guess this would translate to `Quaterniond(std::vector<double>)` right? I am almost certain then that the order on this is `{x, y, z, w}` as Eigen stores the coeffs in this order.

Reply:
this is list initialisation of the return type. Calling `return {a, b, c}` for a return type `Foo` is the same as `return Foo(a, b, c)`. It simply avoids repeating the type in the function body, which is tidier in some cases (like using auto when the type is explicitly given on the right hand side).

The coefficient order therefore is correct, and the tests for CartesianPose etc show that to be true.

Reply:
ok my bad never used that. Gosh so many new stuff to learn...

Reply:
thats very cool